include/mapbox/variant.hpp

```diff
@@ -1023,6 +1023,32 @@ ResultType const& get_unchecked(T const& var)
 {
     return var.template get_unchecked<ResultType>();
 }
+
+// variant_alternative
+template <std::size_t Index, typename...Types>
+struct variant_alternative {};
+
+template <std::size_t Index, typename First, typename...Types>
+struct variant_alternative<Index, First, Types...>
+{
+    using type = typename std::conditional<(sizeof...(Types) == Index),
+                                           First,
+                                           typename variant_alternative<Index, Types...>::type>::type;
+};
+
+template <std::size_t Index>
+struct variant_alternative<Index>
+{
+    using type = void;
+};
```

Comment:
Why do we need this with `type = void`?

Reply:
right, here is alternative implementation 3449d00cf525d8ef98cee0f4a276e2928398a8f9